crates/voicevox_core/src/publish.rs

```diff
@@ -1011,6 +1080,197 @@ mod tests {
+                assert_eq!(mora.vowel(), vowel);
+                // NOTE: 母音の長さが必ず非ゼロになるテストケースを想定している
+                assert_ne!(
+                    mora.vowel_length(),
+                    &0.,
+                    "expected mora.vowel_length is not 0.0, but got 0.0."
+                );
+            }
+        }
+    }
+
+    #[rstest]
+    fn mora_length_works() {
+        let core = VoicevoxCore::new_with_mutex();
+        core.lock()
+            .unwrap()
+            .initialize(InitializeOptions {
+                acceleration_mode: AccelerationMode::Cpu,
+                load_all_models: true,
+                open_jtalk_dict_dir: Some(OPEN_JTALK_DIC_DIR.into()),
+                ..Default::default()
+            })
+            .unwrap();
+
+        let accent_phrases = core
+            .lock()
+            .unwrap()
+            .accent_phrases("これはテストです", 0, AccentPhrasesOptions { kana: false })
+            .unwrap();
+
+        let modified_accent_phrases = core
+            .lock()
+            .unwrap()
+            .mora_length(1, &accent_phrases)
+            .unwrap();
+
+        // NOTE: 一つでも母音の長さが変わっていれば、動作しているとみなす
+        for (before_phrase, after_phrase) in
+            accent_phrases.iter().zip(modified_accent_phrases.iter())
+        {
+            for (before_mora, after_mora) in before_phrase
+                .moras()
+                .iter()
+                .zip(after_phrase.moras().iter())
+            {
+                if before_mora.vowel_length() != after_mora.vowel_length() {
+                    return;
+                }
+            }
+        }
+        panic!("mora_length() does not work: all mora.vowel_length() are same.");
+    }
+
+    #[rstest]
+    fn mora_pitch_works() {
+        let core = VoicevoxCore::new_with_mutex();
+        core.lock()
+            .unwrap()
+            .initialize(InitializeOptions {
+                acceleration_mode: AccelerationMode::Cpu,
+                load_all_models: true,
+                open_jtalk_dict_dir: Some(OPEN_JTALK_DIC_DIR.into()),
+                ..Default::default()
+            })
+            .unwrap();
+
+        let accent_phrases = core
+            .lock()
+            .unwrap()
+            .accent_phrases("これはテストです", 0, AccentPhrasesOptions { kana: false })
+            .unwrap();
+
+        let modified_accent_phrases = core.lock().unwrap().mora_pitch(1, &accent_phrases).unwrap();
+
+        // NOTE: 一つでも音高が変わっていれば、動作しているとみなす
+        for (before_phrase, after_phrase) in
+            accent_phrases.iter().zip(modified_accent_phrases.iter())
+        {
+            for (before_mora, after_mora) in before_phrase
+                .moras()
+                .iter()
+                .zip(after_phrase.moras().iter())
+            {
+                if before_mora.pitch() != after_mora.pitch() {
+                    return;
+                }
+            }
+        }
+        panic!("mora_pitch() does not work: all mora.pitch() are same.");
```

Comment:
ここら辺、共通化できそうだったんですけどやり方が判らなかったのでそのままにしました。

Reply:
私が書くとしたらこんな感じですかね... あとはcrates/downloadで使っているitertoolsを持って来て[`itertools::zip_eq`](https://docs.rs/itertools/0.10/itertools/fn.zip_eq.html)を使った方がいいかもしれません(仮に空の`Vec`が返されたといったときに、テストが通ってしまうので)

```suggestion
        // NOTE: 一つでも音高が変わっていれば、動作しているとみなす
        if !any_mora_param_changed(&accent_phrases, &modified_accent_phrases, MoraModel::pitch) {
            panic!("mora_pitch() does not work: all mora.pitch() are same.");
        }

        use std::iter;

        fn any_mora_param_changed<T: PartialEq>(
            before: &[AccentPhraseModel],
            after: &[AccentPhraseModel],
            param: fn(&MoraModel) -> &T,
        ) -> bool {
            iter::zip(before, after)
                .flat_map(move |(before, after)| iter::zip(before.moras(), after.moras()))
                .any(|(before, after)| param(before) == param(after))
        }
```

Reply:
なるほど。それを使うように置き換えてみました。accent_phrasesが空になることは上のaccent_phrases_worksで検証されてるので大丈夫だと思います。

Reply:
<https://github.com/VOICEVOX/voicevox_core/actions/runs/4846347684/jobs/8635834816>
3つとも間違ったパラメータを比較を比較しているんじゃないかと...

Reply:
ほんとですね。ありがとうございます